docs/CodingGuidelines.md

```diff
@@ -0,0 +1,174 @@
+
+# Coding guidelines
+
+Guidelines that code authors and code reviewers are expected to adhere to in the development of IoT App Kit.
+
+## General guidelines
+
+1. Documentation must be written in a style that is consistent with the rest of the documentation (based on a AWS documentation style)  
+   Compare writing to other written examples within https://docs.aws.amazon.com/iot-sitewise/ as a primary source on how to write the documentation, as well as cross-referencing other documentation found within https://github.com/awslabs/iot-app-kit/tree/main/docs.
+
+1. Ensure that all documentation is correctly referenced in the table of contents, which can be found at https://github.com/awslabs/iot-app-kit/blob/main/README.md.
+
+   In-depth guidelines on documentation: https://alpha-docs-aws.amazon.com/awsstyleguide/latest/styleguide/awsstyleguide.pdf#aws-implementation-guides. This can be a useful reference, but you are not expected to read through and be 100% adherent to these guidelines.
+1. Components must have high quality testing, including integration tests which utilize the component in the way a user does. An example of a good example integration test can be found in the [@iot-app-kit/source-iotsitewise package](https://github.com/awslabs/iot-app-kit/blob/main/packages/source-iotsitewise/src/time-series-data/data-source.spec.ts#L562-L603)
```

Comment:
Thoughts on combining the first two items? Both a documentation related.

Reply:
I'm trying to avoid making long guidelines with many 'rules' as than things get lost. but I agree w/ your idea to add some additional organization so things are easier to find and there could be a 'documentation' section